SitumWayfinding/Classes/Positioning/PositioningViewController.swift

```diff
@@ -1308,4 +1316,63 @@ extension PositioningViewController {
     }
 }
 
-
+extension PositioningViewController {
+    func initializeFloorChangerMarker() {
+        let icon = self.scaledImage(
+            image: UIImage(named: "change_floor")!,
+            scaledToSize: CGSize(width: 40.0, height: 40.0)
+        )
+        let markerView = UIImageView(image: icon)
+        self.changeOfFloorMarker.iconView = markerView
+        self.changeOfFloorMarker.position = CLLocationCoordinate2D(latitude: 0, longitude: 0)
+        resetFloorMarker()
+    }
+    
+    func showFloorChangeMarker(position: CLLocationCoordinate2D) {
+        self.changeOfFloorMarker.position = position
+        self.changeOfFloorMarker.map = self.mapView
+    }
+    
+    func scaledImage(image:UIImage, scaledToSize newSize:CGSize) -> UIImage{
+        UIGraphicsBeginImageContextWithOptions(newSize, false, 0.0)
+        image.draw(in: CGRect(x: 0, y: 0, width: newSize.width, height: newSize.height))
+        let newImage:UIImage = UIGraphicsGetImageFromCurrentImageContext()!
+        UIGraphicsEndImageContext()
+        return newImage
+    }
+    
+    func changePositionFloorMarkerOrResetFloorMarker(segments: Array<SITRouteSegment>, selectedFloor: SITFloor) {
+        resetFloorMarker()
+        
+        if segments.count > 1 {
```

Comment:
Posiblemente o que teñas que facer para que quede esto máis limpo é primerio buscar o segment do piso que queres pintar e unha vez que o teñas crear unha funcion para pintar o cambio de planta so con ese segment
Poque aquí o que estas tentado facer é buscalo e unha vez que o tes pintalo, pero como o fas todo a vez o código queda bastante lioso